Fix roles header behavior + test

bengerman13 opened the pull request:
## Changes proposed in this pull request:

- Fix roles header behavior + test
  `pytest` helpfully flagged the current test as a no-op, because we're essentially asserting that a non-empty tuple is truthy. I put the test back to its behavior before it became a no-op and fixed the code to match. I am not 100% sure whether there should be quotes in the header (`"user", "org-1", "org-2"`) or not (`user, org-1, org-2`), but all the other headers have quotes so I'm guessing that's correct, but Jason added logic clearly meant to strip the quotes so I'm not sure. 

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations

Testing our auth logic better is... better.

markdboyd commented:
I have no issue with these changes, but want @JasonTheMain to comment since #115 he had to re-implement some quoting logic for the header values. We should also be able to test these changes by running the proxy locally and seeing if things work

bengerman13 commented:
I was able to test locally and found that we need the roles unqouted, so I adjusted that. 

I also found a bug in how we were creating our dataclass, so I fixed that too
